### ISSUE TITLE
Only propagate one device device update per session

### ISSUE THREAD

**Issue by postlund** [postlund/pyatv#1265]
**Short feature/function description**
It's possible to get device updates when the connection to a protocol is lost (intentionally or not) via `interface.DeviceListener`. There's a logic problem in the implementation however. All protocols (supporting device updates) will call the methods in said interface whenever a connection is lost. This means that multiple calls can be made from different protocols, which is not the expected behavior. As only MRP and DMAP implement this right now, it's not really a problem (they are normally not used in conjunction with each other).

Another hole here is that when one protocol disconnects, all the other ones should be disconnected as well. This does not happen right now, it's up to the listener to call `close` and clean up.

So, to summarize, this is the expected behavior when a connection is lost:

- [x] A protocol calls `connection_lost` which is forwarded to the actual listener

- [x] All protocols gets disconnected, not yielding any additional calls to `interface.DeviceListener`

**What needs to be done?**
* `interface.StateProducer` must be able to block calls of more than one call is made
* `interface.FacadeAppleTV` must detect when a protocol is disconnected and disconnect all other protocols

**Is this a breaking change?**
No, it should be backwards compatible

**Anything else worth knowing?**
Both Companion and AirPlay are getting persistent connections, so this becomes more important than ever

**Comment by postlund** [comment]
Part of #1263.
